lib/stat_tracker.rb

```diff
@@ -59,4 +59,73 @@ def read_game_teams_csv(location_path)
       end
       game_team_data
    end
+
+   def highest_total_score
+      highest_score_game = @data_games.max_by do |game|
+          game.away_goals + game.home_goals
+      end
+      highest_score_game.away_goals + highest_score_game.home_goals
+   end
+
+   def lowest_total_score
+      lowest_score_game = @data_games.min_by do |game|
+         game.away_goals + game.home_goals
+     end
+     lowest_score_game.away_goals + lowest_score_game.home_goals
+   end
+
+   def percentage_home_wins
+      home_wins = @data_games.find_all do |game|
+         game.home_goals > game.away_goals
+      end
+      calculate_percentage(home_wins.count , @data_games.count)
+   end
+
+   def percentage_visitor_wins
+      away_wins = @data_games.find_all do |game|
+         game.away_goals > game.home_goals
+      end
+      calculate_percentage(away_wins.count , @data_games.count)
+   end
+
+   def percentage_ties
+      ties = @data_games.find_all do |game|
+         game.away_goals == game.home_goals
+      end
+      calculate_percentage(ties.count , @data_games.count)
+   end
+
+   def count_of_games_by_season
+      @data_games.each_with_object(Hash.new(0)) {|game , hash| hash[game.season] += 1}
+   end
+
+   def average_goals_per_game
+      total_goals = 0
+      @data_games.each do |game|
+         total_goals += (game.away_goals + game.home_goals)
+      end
+      (total_goals.to_f / @data_games.count).round(2)
+   end
+
+   def average_goals_by_season
+      goals_by_season = {}
+      @data_games.each do |game|
+         total_goals = game.away_goals + game.home_goals
+         if goals_by_season.key?(game.season)
```

Comment:
@IgorrMagalhaess could we refactor by just assigning the key for goals_by_season hash to game.season and the value would be total goals? (goals_by_season[game.season] = total_goals) Just a bit confused on why the if/else statement is needed here.

Reply:
We can but if we do that we have to do memoization before creating a new key if that key is not found. I can explain this better in our stand_up meeting today!
in this case would be:
`goals_by_season[game.season] ||= 0`
`goas_by_season[game.season] += total_goals`

Or we can initialize our hash in the variable `goals_by_season` with a value of '0'.